decred/decred/dcr/account.py

```diff
@@ -1469,7 +1486,9 @@ def generateGapAddresses(self):
             list(str): Newly generated addresses.
         """
         if self.extPub is None:
-            log.warning("attempting to generate gap addresses on a closed account")
+            raise DecredError(
+                "attempting to generate gap addresses on a closed account"
+            )
```

Comment:
If there's no external branch public extended key (what a mouthful :-) ) the method cannot continue.